ansible_base/authentication/models/authenticator.py

```diff
@@ -39,8 +41,8 @@ def save(self, *args, **kwargs):
 
         if not self.slug:
             self.slug = generate_authenticator_slug(self.type, self.name)
-            # TODO: What happens if computed slug is not unique?
-            # You would have to create an adapter with a name, rename it and then create a new one with the same name
+            if Authenticator.objects.filter(slug=self.slug).count() > 0:
+                self.slug = generate_authenticator_slug(self.type, self.name, secrets.token_hex(2))
```

Comment:
I think we should probably up the number of bytes on the slug suffix to at least 8 bytes if we do not want to loop and recheck for collisions. 1 in 65000 chance of collision is not bad, but I feel like we should probably reduce the collision odds by a little more.

Reply:
Hi Brennan, thanks for that.  Forgot to comment back. :)